app/components/channel_list/categories/body/category_body.tsx

```diff
@@ -12,50 +12,45 @@ import ChannelListItem from './channel';
 import type CategoryModel from '@typings/database/models/servers/category';
 
 type Props = {
-    currentChannelId: string;
     sortedChannels: ChannelModel[];
-    hiddenChannelIds: Set<string>;
     category: CategoryModel;
     limit: number;
 };
 
 const extractKey = (item: ChannelModel) => item.id;
 
-const CategoryBody = ({currentChannelId, sortedChannels, category, hiddenChannelIds, limit}: Props) => {
+const CategoryBody = ({sortedChannels, category, limit}: Props) => {
     const ids = useMemo(() => {
-        let filteredChannels = sortedChannels;
-
-        // Remove all closed gm/dms
-        if (hiddenChannelIds.size) {
-            filteredChannels = sortedChannels.filter((item) => item && !hiddenChannelIds.has(item.id));
-        }
+        const filteredChannels = sortedChannels;
 
         if (category.type === DMS_CATEGORY && limit > 0) {
             return filteredChannels.slice(0, limit - 1);
         }
         return filteredChannels;
-    }, [category.type, limit, hiddenChannelIds, sortedChannels]);
+    }, [category.type, limit, sortedChannels]);
 
-    const ChannelItem = useCallback(({item}: {item: ChannelModel}) => {
```

Comment:
If we re-create the renderItem, the whole list will re-render. To avoid this, we have left the ChannelListItem handle its own activeness. This may generate some more processing in terms of observables being fired, but I expect less trouble than re-rendering the complete list.